src/auth/googleauth.ts

```diff
@@ -126,23 +123,36 @@ export class GoogleAuth {
   }
 
   /**
-   * Obtains the default project ID for the application.
-   * @param callback Optional callback
-   * @returns Promise that resolves with project Id (if used without callback)
+   * THIS METHOD HAS BEEN DEPRECATED.
+   * It will be removed in 3.0.  Please use getProjectId instead.
    */
   getDefaultProjectId(): Promise<string>;
   getDefaultProjectId(callback: ProjectIdCallback): void;
   getDefaultProjectId(callback?: ProjectIdCallback): Promise<string|null>|void {
+    messages.warn(messages.DEFAULT_PROJECT_ID_DEPRECATED);
     if (callback) {
-      this.getDefaultProjectIdAsync()
-          .then(r => callback(null, r))
-          .catch(callback);
+      this.getProjectIdAsync().then(r => callback(null, r)).catch(callback);
```

Comment:
Egads 🙀!

While you're here, fix the `promise.then(f(cb)).catch(cb)` land-mine, or open a bug.

Reply:
Yeah, ooops :) Filed #403 to track.